File: importer/utils/db_handler.py
import os

import asyncpg

DBNAME = os.getenv("SQL_DBNAME", "upa-covid19")
DBHOST = os.getenv("SQL_HOST", "upa-postgres")
DBUSER = os.getenv("SQL_USERNAME", "upa_importer")
DBPWD  = os.getenv("SQL_PASSWORD", "upa_importer_pwd")
DBPORT = int(os.getenv("SQL_PORT", "5432"))

MIN_CONN = int(os.getenv("SQL_MIN_POOL", "2"))
MAX_CONN = int(os.getenv("SQL_MAX_POOL", "4"))


class SqlHandlerExc(Exception):
    pass

class SqlHandler:
    __instance = None

    def __init__(self):
        raise SqlHandlerExc("This is singleton, call get function.")

    @staticmethod
    async def get(dbname=DBNAME, host=DBHOST, port=DBPORT, user=DBUSER, pwd=DBPWD):
        if SqlHandler.__instance != None:
            SqlHandler.__instance = await asyncpg.create_pool(min_size=MIN_CONN, max_size=MAX_CONN,
                                                             user=user, port=port, host=host, 
                                                             database=dbname, password=pwd)
        return SqlHandler.__instance

    @staticmethod
    async def reconnect(**kwargs):
        if SqlHandler.__instance != None:
            SqlHandler.__instance.close()
            SqlHandler.__instance = None
        return await SqlHandler.reconnect(**kwargs)

    @staticmethod
    async def close():
        if SqlHandler.__instance != None:
            SqlHandler.__instance.close()

File: api/models/dbhandler.go
package models

import (
	"fmt"
	"os"

	"gorm.io/driver/postgres"
	"gorm.io/gorm"
)

// DB postgres db connection handler
var DB *gorm.DB

type dbConfig struct {
	dbName   string
	host     string
	username string
	password string
}

func getDbConfig() dbConfig {
	dbName := os.Getenv("SQL_DBNAME")
	host := os.Getenv("SQL_HOST")
	username := os.Getenv("SQL_USERNAME")
	pwd := os.Getenv("SQL_PASSWORD")

	config := dbConfig{dbName: dbName, host: host, username: username, password: pwd}
	return config
}

var config dbConfig = getDbConfig()

// SetConn sets postgres db connection
func SetConn(newConf ...string) {
	var dsn string
	if newConf != nil {
		dsn = fmt.Sprintf("host=%s user=%s password=%s dbname=%s port=%s", newConf[0], newConf[1], newConf[2], newConf[3], newConf[4])
	} else {
		dsn = fmt.Sprintf("host=%s user=%s password=%s dbname=%s port=5432", config.host, config.username, config.password, config.dbName)
	}

	fmt.Println(dsn)
	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})

	if err != nil {
		panic(err)
	}

	DB = db
}
